response.go

```
@@ -11,6 +11,21 @@ func Arguments() []string {
 	return os.Args[1:]
 }
 
+// NormalizedArguments re-normalizes the user arguments provided via Alfred.
+// This isn't necessary for every workflow, specifically only when you're working with special characters.
+// For more info on this topic, please refer to this thread: http://www.alfredforum.com/topic/2015-encoding-issue/
+func NormalizedArguments() (normalizedArgs []string, err error) {
+	for _, e := range Arguments() {
+		var normalizedElement string
+		normalizedElement, err = Normalize(e)
+		if err != nil {
+			return
```

So, if normalizing on out of potential multiple arguments fails, the others after that argument are also not part of the arguments anymore.. I don't like this behavior. I would expect that either I get an empty array in error case or I get all elements and the ones that failed are not normalized or I get all arguments except the ones that couldn't be normalized.

---

I would've expected to check the error on the other side and if present, ignore the partially normalized input altogether. But you're right, people can also choose to ignore the error and will get stumped as to why they're only getting a part of the results.

---

Yeah you are right, but let's make it also robust enough for people that don't do error handling :D I changed it in 1735ad2 to use continue, so that not normalized elements are now not part of the result anymore.